recipes/google-cloud-cpp-compute/build.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -euo pipefail
+
+export OPENSSL_ROOT_DIR=$PREFIX
+
+if [[ "${target_platform}" == osx-* ]]; then
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+  CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+fi
+
+  echo "$(date -u '+%Y-%m-%dT%H:%M:%SZ'): Building compute"
+  cmake ${CMAKE_ARGS} \
+      -GNinja -S . -B .build/compute \
```

Comment:
Can we be more explicit about which folders exist already, and avoid names with a leading dot? And also use the same names consistently between building and testing?

This might all seem a bit unnecessary, but it's possible (though not recommended) to pipe some file from the build stage through to the testing stage (and similarly, it's possible that the sources already contain a `./build` folder).

By consistently creating the the respective folders in the scripts, we demonstrate that we're not relying on either of that. What I'm thinking is something along the lines of:

```
mkdir -p build/compute
cd build/compute  # or pushd, if you prefer

cmake -G Ninja \
    ${CMAKE_ARGS} \
    -D... \
    -D... \
    ../..  # or $SRC_DIR

cmake --build .
```

Reply:
> Can we be more explicit about which folders exist already,

I am not sure what you mean by this.

> and avoid names with a leading dot?

Happy to do that, though they are harmless.  AFAIK none of the Windows filesystems have limitations around a leading dot.

> And also use the same names consistently between building and testing?

Happy to do that too.

> This might all seem a bit unnecessary, but it's possible (though not recommended) to pipe some file from the build stage through to the testing stage (and similarly, it's possible that the sources already contain a `./build` folder).

It seems more likely that the sources would contain a `./build` directory than a `./.build` folder, but meh.

> By consistently creating the the respective folders in the scripts, we demonstrate that we're not relying on either of that. What I'm thinking is something along the lines of:
> 
> ```
> mkdir -p build/compute
> cd build/compute  # or pushd, if you prefer
> 
> cmake -G Ninja \
>     ${CMAKE_ARGS} \
>     -D... \
>     -D... \
>     ../..  # or $SRC_DIR
> 
> cmake --build .
> ```

`cmake -S <path to source> -B <path to build>` is a portable way to create the build directory and define the source directory. It is documented since CMake 3.13, and works since at least CMake 3.5.  Why use more commands to do the same thing that CMake already does?  It is not more explicit, or less explicit, just maybe less familiar thank explicit `mkdir` calls?